clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -1089,29 +1106,61 @@ private long sendEligibleCalls(long now) {
                     continue;
                 }
                 Node node = entry.getKey();
+                if (!callsInFlight.getOrDefault(node.idString(), Collections.emptyList()).isEmpty()) {
+                    log.trace("Still waiting for other calls to finish on node {}.", node);
+                    nodeReadyDeadlines.remove(node);
+                    continue;
```

Comment:
Maybe I am missing something here. Why do we need to continue? Do we not support multiple in-flight requests? If not, then why does `callsInFlight` map to a list?

Reply:
Notice that we set `maxInFlightRequestsPerConnection` to 1 when constructing the `NetworkClient`.  We don't support sending multiple requests to a single node on a single connection in `AdminClient`.  I think we could add this support, but we'd have to check how the server handled it since we've never done it before.  Maybe there should be a JIRA.

Also, if we do choose to add this support for multiple outstanding requests per node per socket we'd need some way to distinguish between "waiting for a chance to use this connection" from "waiting for this connection to be opened" in NetworkClient.  Currently ready just returns a boolean, which isn't enough information to distinguish these two cases.  We could probably add a new method that returned an enum or something.

Reply:
Oh, and on the question of why inflight requests has a list, it was done for future-proofing

Reply:
ok.  I changed it to just `Map<String, Call>`